core/src/main/scala/chisel3/VerificationStatement.scala

```diff
@@ -65,8 +97,35 @@ object assert {
     compileOptions: c.Tree
   ): c.Tree = {
     import c.universe._
-    val apply_impl_do = symbolOf[this.type].asClass.module.info.member(TermName("_applyWithSourceLine"))
-    q"$apply_impl_do($cond, ${getLine(c)}, _root_.scala.Some($message), ..$data)($sourceInfo, $compileOptions)"
+    val apply_impl_do = symbolOf[this.type].asClass.module.info.member(TermName("_applyWithSourceLinePrintable"))
+    q"$apply_impl_do($cond, ${getLine(c)},_root_.scala.Some(_root_.chisel3.Printable.pack($message,..$data)))($sourceInfo, $compileOptions)"
+  }
+
+  /** Used by our macros. Do not call directly! */
```

Comment:
You could use the same or create a new  `@group` to make this show up in a 'Here be dragons" section of the API docs:

```
  /** @group SourceInfoTransformMacro */
```

This renders in the API docs like so:

<img width="955" alt="Screen Shot 2022-05-13 at 12 52 18 PM" src="https://user-images.githubusercontent.com/17858596/168380005-b3799cb8-e103-433b-925e-9274e0ac17ff.png">

https://github.com/chipsalliance/chisel3/blob/83eeda2efebb1e3e3d66365d388f5de627766d47/macros/src/main/scala/chisel3/SourceInfoDoc.scala#L22

Reply:
Yeah this would be a good thing to do.

Reply:
Actually, this gave me an idea, we should create `object internal` in each of these objects and then put all of the `_` methods in there. That way it's clear that there's are internal things that users shouldn't use.

The current ScalaDoc for `object assert` is a mess:
![Screen Shot 2022-05-13 at 1 16 02 PM](https://user-images.githubusercontent.com/8584418/168383147-6bcd5570-2f06-439e-b920-b4e80b99f655.png)

Putting the internal methods in `object internal` helps a ton:

![Screen Shot 2022-05-13 at 1 16 39 PM](https://user-images.githubusercontent.com/8584418/168383234-4eaca272-be43-410b-a69d-ce7a380c3403.png)

Since it would be binary incompatible to just move them all, we may still want to do the grouping though because we'll need to deprecate a few of the methods before moving them.

Reply:
@mwachs5  @jackkoenig   - thanks for the pointers / tips - but just for clarity what do you feel is a MUST for this PR? Obviously long term we should incorporate all of these suggestions!

Reply:
i think we can easily add a @group to this PR